crates/aide/src/axum/mod.rs

```diff
@@ -439,30 +439,29 @@ where
 
         let _ = transform(TransformOpenApi::new(api));
 
-        let needs_reset =
-            in_context(|ctx| {
-                if !ctx.extract_schemas {
-                    return false;
-                }
+        let needs_reset = in_context(|ctx| {
+            if !ctx.extract_schemas {
+                return false;
+            }
 
-                let components = api.components.get_or_insert_with(Default::default);
-                components
-                    .schemas
-                    .extend(ctx.schema.take_definitions().into_iter().map(
-                        |(name, json_schema)| {
-                            (
-                                name,
-                                SchemaObject {
-                                    json_schema,
-                                    example: None,
-                                    external_docs: None,
-                                },
-                            )
-                        },
-                    ));
-
-                true
-            });
+            let components = api.components.get_or_insert_with(Default::default);
+            components
+                .schemas
+                .extend(ctx.schema.take_definitions(true).into_iter().map(
+                    |(name, json_schema)| {
+                        (
+                            name,
+                            SchemaObject {
+                                json_schema: json_schema.try_into().expect("Invalid schema"),
```

Comment:
This one is a bit tricky. I don't want to introduce panics, but at the same time the current library design doesn't allow for functions as this one to fail. As i don't want to introduce any breaking changes to how this library is used, this seems to be the only way.

Reply:
I think it's fine to panic in the beginning. We can bump aide major version as schemars 0.8 -> 0.9 is already a breaking change

Reply:
Can you explain why this could fail now when it couldn't before? What is the `try_into` converting here?

Reply:
Since schemars 0.9.0 now just wrapps everything around a `serde_json::Value` we don't get a strictly defined `Schema` struct anymore, as we used to. What's happening here, is that we're converting the `json_schema ` (which is of type `serde_json::Value`) into a `Schema` unit struct (defined in the lib as `Schema(serde_json::Value)`), which implements a TryFrom<serde_json::Value> as it does some internal validation. 
Usually this should never panic, as that would require the `json_schema` variable to be malformed, and since that's just coming from another `Schema`, it's probably not even possible.